samples/create-buffers-around-points/src/main/java/com/esri/arcgismaps/sample/createbuffersaroundpoints/components/CreateBuffersAroundPointsViewModel.kt

```diff
@@ -0,0 +1,274 @@
+    private val bufferGraphicsOverlay by lazy {
+        val bufferOutline = SimpleLineSymbol(
+            style = SimpleLineSymbolStyle.Solid,
+            color = Color.green,
+            width = 3f
+        )
+        val bufferFill = SimpleFillSymbol(
+            style = SimpleFillSymbolStyle.Solid,
+            color = Color.fromRgba(r = 255, g = 255, b = 0, a = 153),
+            outline = bufferOutline
+        )
+        GraphicsOverlay().apply { renderer = SimpleRenderer(symbol = bufferFill) }
+    }
+
+
+    // Configure tapped points overlay renderer (small red circles)
+    private val tappedPointsGraphicsOverlay by lazy {
+        val tapSymbol = SimpleMarkerSymbol(
+            style = SimpleMarkerSymbolStyle.Circle,
+            color = Color.red,
+            size = 10f
+        )
+        GraphicsOverlay().apply { renderer = SimpleRenderer(symbol = tapSymbol) }
+    }
+   
+    val graphicsOverlays: List<GraphicsOverlay>
+        get() = listOf(boundaryGraphicsOverlay, bufferGraphicsOverlay, tappedPointsGraphicsOverlay)
+
+    // Keep last tapped point when requesting radius input
+    private var lastTappedPoint: Point? = null
+
+    // List of (point, radiusInMapUnits)
+    private val bufferPoints: MutableList<Pair<Point, Double>> = mutableListOf()
+
+
+    private val _statusText = MutableStateFlow("Tap on the map to add buffers.")
+    val statusText = _statusText.asStateFlow()
+
+    private val _isInputDialogVisible = MutableStateFlow(false)
+    val isInputDialogVisible = _isInputDialogVisible.asStateFlow()
+
+    // Controls whether buffers should be union
+    private val _shouldUnion = MutableStateFlow(false)
+    val shouldUnion = _shouldUnion.asStateFlow()
+
+    init {
+        viewModelScope.launch {
+            // Load the map; report any failure
+            arcGISMap.load().onFailure { messageDialogVM.showMessageDialog(it) }
+        }
+    }
+
+    /**
+     * Called by the MapView when the user taps the map. If the tap is within the valid boundary,
+     * request a buffer radius input from the UI. Otherwise update status to indicate out-of-bounds.
+     */
+    fun onMapTapped(mapPoint: Point) {
+        val contains = GeometryEngine.contains(boundaryPolygon, mapPoint)
+        if (!contains) {
+            _statusText.value = "Tap within the boundary to add buffer."
+        }
+
+        // Store the tapped point and request input dialog
+        lastTappedPoint = mapPoint
+        _isInputDialogVisible.value = true
+        _statusText.value = "Enter buffer radius (miles) for the tapped location"
+    }
+
+    /**
+     * Submit the radius (in miles) provided by the user. Converts miles to the map's linear units
+     * (the state plane spatial reference uses US feet) then creates the buffer and updates the overlays.
+     */
+    fun submitRadiusMiles(radiusMiles: Double) {
+        viewModelScope.launch {
+            val point = lastTappedPoint
```

Comment:
To avoid `!!` below: 

```suggestion
            val point = lastTappedPoint ?: return
```